src/bridge-deployer/src/canister_ids/db.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::HashMap;
+
+use candid::Principal;
+use serde::{Deserialize, Serialize};
+
+use super::principals::CanisterPrincipal;
+use super::CanisterType;
+use crate::contracts::EvmNetwork;
+
+/// canister_ids.json db
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct CanistersDb {
+    #[serde(flatten)]
+    canisters: HashMap<CanisterType, CanisterPrincipal>,
```

Comment:
Sorry, I still don't understand why there's a CanisterPrincipal here

Reply:
I updated the name to CanisterPrincipalS, so it's association between the bridge type and its principals based on network.